### PR TITLE
nickel: disable checks on Darwin

### PR DESCRIPTION
**Pull request by YorikSar**
###### Description of changes

Workaround to address build issue introduced in 1.1.1 update.
Upstream issue: https://github.com/tweag/nickel/pull/1454

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/interpreters/nickel/default.nix [hunk 1]
```diff
@@ -24,6 +24,9 @@ rustPlatform.buildRustPackage rec {
     python3
   ];
 
+  # Disable checks on Darwin because of issue described in https://github.com/tweag/nickel/pull/1454
+  doCheck = !rustPlatform.rust.rustc.stdenv.isDarwin;
```
Comment (felschr) [inline pkgs/development/interpreters/nickel/default.nix:28]:
```suggestion
  doCheck = !stdenv.isDarwin;
```
and add `stdenv` to the arguments list.

**Review by felschr — CHANGES_REQUESTED** [review]

**Review by felschr — CHANGES_REQUESTED** [review]
The same `doCheck` should be added to `nls` as well.

**Comment by YorikSar** [comment]
@felschr I've switched to `stdenv.isDarwin` and added same clause to `nls`.

**Review by felschr — APPROVED** [review]
This looks good to me if the darwin builds succeed.

**Pull request merged**